src/osd/OSD.cc

```diff
@@ -5942,7 +5943,8 @@ OSDService::ScrubJob::ScrubJob(const spg_t& pg, const utime_t& timestamp, bool m
       sched_time += rand() % (int)(g_conf->osd_scrub_min_interval *
```

Comment:
I'll ignore some of the other scrub options like randomization and restricted scrub times.

"In my proposal osd_scrub_max_interval is still used until it is more then pool_scrub_interval, so the user can only make delay more per pool."
That's not how I read the code.  If pool_scrub_interval > 0 the deadline is calculated as the time when pool_scrub_interval has passed instead of when osd_scrub_max_interval has passed.

1. Current code: pool_scrub_interval overrides osd_scrub_max_interval:
Setting pool_scrub_interval below osd_scrub_min_interval, will always scrub at osd_scrub_min_interval.  Otherwise, a given pool can have its own max_interval.  It can scrub before pool_scrub_interval if load is below the threshold.

 2. If you modify both the min and max based on pool_scrub_interval:
The load average will not be considered and scrubbing all occur based on pool_scrub_interval.  That happens anytime min and max are the same because it only tries to scrub after min and ignores the load after max.